tests/composition/pec/test_parameterestimationcomposition.py

```diff
@@ -145,17 +188,29 @@ def test_pec_run_input_formats(inputs_dict, error_msg):
 
 @pytest.mark.composition
 @pytest.mark.parametrize(
-    "opt_method, expected_result",
+    "opt_method, optuna_kwargs, expected_result",
     [
-        ("differential_evolution", expected_differential_evolution),
-        (optuna.samplers.RandomSampler(seed=0), [0.01]),
-        (optuna.samplers.CmaEsSampler(seed=0), [0.01]),
+        ("differential_evolution", None, expected_differential_evolution),
+        (optuna.samplers.RandomSampler(seed=0), None, [0.01]),
+        (optuna.samplers.QMCSampler(seed=0), None, [0.01]),
+        (optuna.samplers.RandomSampler, {'seed': 0}, [0.01]),
+        (optuna.samplers.RandomSampler(), None, None)
+    ],
+    ids=[
+        "differential_evolution",
+        "optuna_random_sampler",
+        "optuna_qmc_sampler",
+        "optuna_random_sampler_with_kwargs",
+        "optuna_random_sampler_no_seed"
     ],
-    ids=["differential_evolution", "optuna_random_sampler", "optuna_cmaes_sampler"],
 )
-def test_parameter_optimization_ddm(func_mode, opt_method, expected_result):
+def test_parameter_optimization_ddm(func_mode, opt_method, optuna_kwargs, expected_result):
     """Test parameter optimization of a DDM in integrator mode"""
 
+    # Dicts are being converted to mappingproxy objects, need to convert to dict.
+    if optuna_kwargs:
+        optuna_kwargs = {k: v for k, v in optuna_kwargs.items()}
```

Comment:
Why is this necessary? Can't optuna handle general mapping type?

Reply:
Ok, I tried to fix this one better as well. It was a bit trickier to track down. It seems that optuna_kwargs argument gets cached in _user_specified_args since its and arg to a function. Then later, a deepcopy operation fails on the PECOptimizationFunction because it can't copy\pickle the mappingproxytype. I added a change in check_user_specified that converts MappingProxyTypes to dicts, does this look like an appropriate fix @kmantel @jvesely

Reply:
Hm, I think I intended to have `_user_specified_args` be deleted after initialization but it doesn't look like it is. It shouldn't be needed after that. Is it just that one attribute that's a problem?

Reply:
This seems to have worked for my test, checking to see if it breaks something else.